stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/SignatureParams.java

```diff
@@ -0,0 +1,116 @@
+
+import java.util.List;
+
+/**
+ * This class holds the resource signature parameters.
+ *
+ * @since 0.963.0
+ */
+public class SignatureParams {
+
+    private HttpResource resource;
+    private final List<ParamDetail> paramDetails;
+    private ParamDetail entityBody;
+    private List<ParamDetail> pathParams;
+    private int paramCount = 2;
+
+    public SignatureParams(HttpResource resource, List<ParamDetail> paramDetails) {
+        this.resource = resource;
+        this.paramDetails = paramDetails;
+    }
+
+    public void validate() {
+
+        if (paramDetails.size() < 2) {
+            throw new BallerinaConnectorException("resource signature parameter count should be more than two");
+        }
+        if (!isValidResourceParam(paramDetails.get(0), HttpConstants.CONNECTION)) {
+            throw new BallerinaConnectorException("first parameter should be of type - "
+                    + HttpConstants.PROTOCOL_PACKAGE_HTTP + ":" + HttpConstants.CONNECTION);
+        }
+        if (!isValidResourceParam(paramDetails.get(1), HttpConstants.IN_REQUEST)) {
+            throw new BallerinaConnectorException("second parameter should be of type - "
+                    + HttpConstants.PROTOCOL_PACKAGE_HTTP + ":" + HttpConstants.IN_REQUEST);
+        }
+        if (paramDetails.size() == 2) {
+            return;
+        }
+
+        if (resource.getEntityBodyAttributeValue() == null) {
+            validatePathParam(paramDetails.subList(2, paramDetails.size()));
+        } else {
+            int lastParamIndex = paramDetails.size() - 1;
+            validatePathParam(paramDetails.subList(2, lastParamIndex));
+            validateEntityBodyParam(paramDetails.get(lastParamIndex));
+        }
+    }
+
+    private boolean isValidResourceParam(ParamDetail paramDetail, String varTypeName) {
+        return paramDetail.getVarType().getPackagePath() != null
+                && paramDetail.getVarType().getPackagePath().equals(HttpConstants.PROTOCOL_PACKAGE_HTTP)
+                && paramDetail.getVarType().getName().equals(varTypeName);
+    }
+
+    private void validatePathParam(List<ParamDetail> paramDetails) {
+        for (int i = 0; i < paramDetails.size(); i++) {
+            ParamDetail paramDetail = paramDetails.get(i);
+            if (paramDetail.getVarType().getTag() != TypeTags.STRING_TAG) {
+                throw new BallerinaConnectorException("incompatible resource signature parameter type");
+            }
+            paramCount++;
+        }
+        this.pathParams = paramDetails;
+    }
+
+    private void validateEntityBodyParam(ParamDetail entityBodyParam) {
+        String entityBodyAttributeValue = resource.getEntityBodyAttributeValue();
+        if (!entityBodyAttributeValue.equals(entityBodyParam.getVarName())) {
+            throw new BallerinaConnectorException("expected " + entityBodyAttributeValue +
+                    " as param name instead of " + entityBodyParam.getVarName());
```

Comment:
Shall we rephrase this to something like the following: 
``
"expected '" + entityBodyAttributeValue + "' as param name, but found '" + entityBodyParam.getVarName() + "'"
``

Reply:
Addressed with b98823c.